edward/inferences/inference.py

```diff
@@ -188,7 +188,11 @@ def run(self, logdir=None, variables=None, use_coordinator=True,
     self.initialize(*args, **kwargs)
 
     if logdir is not None:
+      self.logging = True
       self.train_writer = tf.train.SummaryWriter(logdir, tf.get_default_graph())
+      self.summarize = tf.merge_all_summaries()
+    else:
+      self.logging = False
```

Comment:
What about the idea of allowing `initialize` to take `logdir` as a parameter, setting the flag there, and moving [this](https://github.com/blei-lab/edward/pull/337#discussion-diff-87717005R128) (the actual logging) into `update`?